### PR TITLE
Fix support for maxPodsPerNode in unmanaged nodegroups

### LINKED ISSUE #3818: maxPodsPerNode does not work
**What were you trying to accomplish?**

Trying to increase the number of pods running on a node

**What happened?**

I set `maxPodsPerNode: 110` when creating a `t3.medium` nodegroup.

We have the Weave CNI installed (ripped out the crappy AWS CNI)

**How to reproduce it?**

Create a nodegroup like this:

```
  - name: eks-us-west-2-node-group-rabbitmq-medium-v1
    labels:
      type: medium
      workload: rabbitmq
    taints:
      dedicated: rabbitmq:NoSchedule
    desiredCapacity: 1
    minSize: 1
    maxSize: 1
    instanceType: t3.medium
    iam:
      withAddonPolicies:
        cloudWatch: true
        ebs: true
    privateNetworking: true
    tags:
      Environment: UAT
    disableIMDSv1: true
    maxPodsPerNode: 110
    kubeletExtraConfig:
      kubeReserved:
        cpu: "300m"
        memory: "300Mi"
        ephemeral-storage: "1Gi"
      kubeReservedCgroup: "/kube-reserved"
      systemReserved:
        cpu: "300m"
        memory: "300Mi"
        ephemeral-storage: "1Gi"
      evictionHard:
        memory.available: "200Mi"
        nodefs.available: "10%"
```

Then get the node:  `kubectl get nodes -l workload=rabbitmq -o yaml` and observe the capacity is still at 17:

```
  status:
    addresses:
    - address: 10.235.245.153
      type: InternalIP
    - address: ip-10-235-245-153.us-west-2.compute.internal
      type: Hostname
    - address: ip-10-235-245-153.us-west-2.compute.internal
      type: InternalDNS
    allocatable:
      attachable-volumes-aws-ebs: "25"
      cpu: 1400m
      ephemeral-storage: "75150584500"
      hugepages-1Gi: "0"
      hugepages-2Mi: "0"
      memory: 3148920Ki
      pods: "17"
    capacity:
      attachable-volumes-aws-ebs: "25"
      cpu: "2"
      ephemeral-storage: 83873772Ki
      hugepages-1Gi: "0"
      hugepages-2Mi: "0"
      memory: 3968120Ki
      pods: "17"
```

**Logs**

**Anything else we need to know?**

**Versions**

```
$ eksctl version
$ kubectl version
```
eksctl version: `0.49.0`
kubernetes version:  `Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.8-eks-96780e"`

### PR DESCRIPTION
**Pull request by cPu1**
### Description

Fixes support for `maxPodsPerNode` in unmanaged nodegroups

Fixes #3818 

TODO:

- [x] Add integration tests

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/nodebootstrap/userdata.go [hunk 1]
```diff
@@ -139,31 +140,40 @@ func makeKubeletExtraConf(kubeletExtraConf *api.InlineDocument) (cloudconfig.Fil
...
+		"CLUSTER_NAME":   clusterConfig.Metadata.Name,
+		"API_SERVER_URL": clusterConfig.Status.Endpoint,
+		"B64_CLUSTER_CA": base64.StdEncoding.EncodeToString(clusterConfig.Status.CertificateAuthorityData),
+		"NODE_LABELS":    formatLabels(ng.Labels),
+		"NODE_TAINTS":    utils.FormatTaints(np.NGTaints()),
+	}
+
+	if ng.MaxPodsPerNode > 0 {
+		variables["MAX_PODS"] = strconv.Itoa(ng.MaxPodsPerNode)
 	}
```
Comment (Callisto13) [inline pkg/nodebootstrap/userdata.go:154]: Can we add a unit test to check this ends up in userdatas?

Comment (cPu1) [reply pkg/nodebootstrap/userdata.go:154]: Done.

**Review by Callisto13 — APPROVED** [review]
Cool, once the integration tests pass this is good to merge and I will release 👍

**Comment by Callisto13** [comment]
Are those integration tests also running the Bottlerocket fix?

**Comment by cPu1** [comment]
> Are those integration tests also running the Bottlerocket fix?

They're not because it isn't merged. I have tested it manually though.

**Comment by Callisto13** [comment]
This one? https://github.com/weaveworks/eksctl/pull/3820

**Comment by cPu1** [comment]
> This one? #3820

Yeah, it got merged after I triggered the integration tests. I have triggered the tests again after rebasing.

**Comment by Callisto13** [comment]
Cool thank you!

**Pull request merged**